Tasks/AzureRmWebAppDeploymentV4/operations/AzureAppServiceUtility.ts

```diff
@@ -253,17 +268,39 @@ export class AzureAppServiceUtility {
 
         if(!!releaseDefinitionId) {
             // Task is running in Release
-            let buildDefintionId = tl.getVariable("build.definitionId");
+            tl.debug("Artifact Source Alias is: "+ artifactAlias);
+            let artifactType = tl.getVariable(`release.artifacts.${artifactAlias}.type`);
```

Comment:
I assume when releaseDefinitionId is non-null then artifactAlias will be a valid value

Reply:
package path can be some local folder path when task is running on local agent.

Reply:
If that is the case, you should check that if (releaseDefinitionId && artifactAlias) in the containing if condition here!

Reply:
yes. That's right. artifact condition is missing in if, it should be there. same condition is there while determining kudu log details.